bugwarrior/services/logseq.py

```diff
@@ -0,0 +1,301 @@
+import logging
+
+import requests
+import typing_extensions
+
+import re
+from datetime import datetime
+
+from bugwarrior import config
+from bugwarrior.services import IssueService, Issue, ServiceClient
+
+log = logging.getLogger(__name__)
+
+
+class LogseqConfig(config.ServiceConfig):
+    service: typing_extensions.Literal["logseq"]
+    host: str = "localhost"
+    port: int = 12315
+    token: str = ""
+    task_state: str = "DOING, TODO, NOW, LATER, IN-PROGRESS, WAIT, WAITING"
+    char_open_link: str = "【"
+    char_close_link: str = "】"
+    char_open_bracket: str = "〈"
+    char_close_bracket: str = "〉"
+    inline_links = True  # set to False to override bugwarrior config
+
+
+class LogseqClient(ServiceClient):
+    def __init__(self, host, port, token, filter):
+        self.host = host
+        self.port = port
+        self.token = token
+        self.filter = filter
+
+        self.headers = {
+            "Authorization": "Bearer " + self.token,
+            "content-type": "application/json; charset=utf-8",
+        }
+
+        self.project = None
+
+    def _datascript_query(self, query):
+        try:
+            response = requests.post(
+                f"http://{self.host}:{self.port}/api",
+                headers=self.headers,
+                json={"method": "logseq.DB.datascriptQuery", "args": [query]},
+            )
+            return self.json_response(response)
+        except requests.exceptions.ConnectionError as ce:
+            log.fatal("Unable to connect to Logseq HTTP APIs server. %s", ce)
+            exit()
```

Comment:
Today I learned there is [a builtin exit function](https://docs.python.org/3.8/library/constants.html#exit). While I'm more used to seeing `sys.exit` or simply raising an exception, I guess this is fine. However, I do think you should specify an exit code because otherwise when it propagates to [SystemExit](https://docs.python.org/3.8/library/exceptions.html#SystemExit) " if it is None, the exit status is zero".
```suggestion
            exit(1)
```